stactools_planet/stactools/planet/planet_item.py

```diff
@@ -196,6 +210,17 @@ def to_stac(self):
                         ]
                     item.ext.eo.set_bands(bands, asset)
 
+            if add_file_info:
+                item.ext.enable('file')
+                checksum = self._file_checksum(href)
+                with rasterio.open(href) as dataset:
+                    # we assume all bands in the image have the same data type
+                    data_type = FileDataType(dataset.dtypes[0])
+                size = os.path.getsize(href)
```

Comment:
`getsize` will only work, but using the name `href` to me implies it should work with remote paths too. Is that intended?

Reply:
I am just the variable that was there before these changes, but you are right, that won't work with a remote file path. We can put that in a try block ad only add those parameters when it's possible.

In any case, I am not sure if href in this case can be a remote path. Isn't the idea of this extension to create a STAC catalog from a planet bundle that you have downloaded from Planet (and is therefore local)?

Reply:
> Isn't the idea of this extension to create a STAC catalog from a planet bundle that you have downloaded from Planet (and is therefore local)? 

I haven't used this module, but I figured it might have also been intended to work with a Planet API URL. From a cursory look, other steps like the `rasterio.open` would work with both local files and remote URLs.